### PR TITLE
Guides screens as Components

### PR DESCRIPTION
**Pull request by 4yki**

### CODE REVIEW

File: view/components/Banner/index.tsx [hunk 1]
```diff
@@ -62,7 +63,7 @@ const Banner = () => {
       source={{ uri: item }}
       style={{
         width: dimensions.width,
-        height: dimensions.height / 6
+        height: BANNER_HEIGHT
```
Comment (GuyAvraham) [inline view/components/Banner/index.tsx:66]: @4yki I remember that I used here the screen height because it needs to look proportional on IPads. 
Why is the change?

File: view/components/GuideModal/crumbs.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+const GuideContent = (props: GuideProps) => {
+  const insets = useSafeArea();
+  const headerHeight = useHeaderHeight();
+
+  switch (props.pageNumber) {
+    case 0:
+      return (
+        <View style={containerStyle(insets.top, 'center', 0)}>
+          <HeaderLogo headerAreaHeight={headerHeight - insets.top} />
```
Comment (GuyAvraham) [inline view/components/GuideModal/crumbs.tsx:30]: note to self: check why is that needed

File: Utils/Localization.ts [hunk 3]
```diff
@@ -133,7 +139,14 @@ i18n.translations = {
       CheckAgain: 'בדוק הרשאות שוב',
       OpenSettings: 'נווט להגדרות המכשיר'
     },
-  },
+    Guide: {
+      First: 'את נמצאת באיזור האישי. בלחיצה על הלוגו תוכלי לחזור למסך הראשי.',
+      Second: 'את נמצאת במסך הראשי, בלחיצה על שמך, הנך מועברת לאזור האישי שלך',
+      Third: 'את נמצאת במסך הראשי, בלחיצה על מרכז הזמנות, הנך מבצעת רכישה חדשה',
+      Fourth:
+        'את נמצאת במסך הראשי, בלחיצה על ייעוץ מקצועי, הנך מועברת לשירות לקוחות'
```
Comment (GuyAvraham) [inline Utils/Localization.ts:147]: how did you wrote the hebrew?

Comment (4yki) [reply Utils/Localization.ts:147]: Just copy/pasted from Adobe XD design. I'm not sure that it's the correct strings, so it would be nice if you correct them

File: view/components/GuideModal/crumbs.tsx (hunk 2 above)
Comment (4yki) [reply view/components/GuideModal/crumbs.tsx:30]: If you have any questions, we can make a short call on monday.

File: view/components/Banner/index.tsx [hunk 4]
```diff
@@ -7,6 +7,7 @@ import { useStateValue } from '../../../Utils/State';
 
 const dimensions = Dimensions.get('window');
 const AUTOPLAY_INTERVAL = 5000;
+export const BANNER_HEIGHT = 112;
```
[the remark this answers is not in the document]
Comment (GuyAvraham) [reply view/components/Banner/index.tsx:10]: now all header sizes are fixed. It will look bad  on androids and tablets. 
Did you touch this code because of what you build on guide tour? 
Or just to improve code?

File: view/pages/Main/MainPage/index.tsx [hunk 5]
```diff
@@ -1,33 +1,56 @@
...
 
 const { botUrl, wpBase1 } = get(getSettings(), 'config');
 
 const MainPage = (props: Props) => {
-  const [{ user }, dispatch] = useStateValue();
+  const [{ user, guideMode }, dispatch] = useStateValue();
+  const [guideModalVisible, setGuideModalVisible] = useState(false);
+  const headerHeight = useHeaderHeight();
+  const dimensions = useSafeArea();
+  const insets: EdgeInsets = useSafeArea();
```
Comment (GuyAvraham) [inline view/pages/Main/MainPage/index.tsx:34]: useSafeArea() is used twice. I fixed

File: view/components/Logo.tsx [hunk 6]
```diff
@@ -1,28 +0,0 @@
-import { Image, StyleSheet, TouchableOpacity, Dimensions } from 'react-native';
-import React from 'react';
-
-const deviceWidth = Dimensions.get('window').width;
-
-const styles = StyleSheet.create({
-  logo: {
-    height: undefined,
-    width: deviceWidth / 2.8,
-    aspectRatio: 1,
```
Comment (GuyAvraham) [inline view/components/Logo.tsx:10]: this one was important. scaling image proportialy

Comment (4yki) [reply view/components/Logo.tsx:10]: We have fixed height on the header, so using this height in the styles of the Logo image container and Logo image itself is enough to maintain aspect ratio, because we have "resizeMode" = "contain". We can even omit setting width prop. I added it in order to have limited clickable area.

File: view/pages/Main/MainPage/index.tsx (hunk 5 above)
Comment (4yki) [reply view/pages/Main/MainPage/index.tsx:35]: yes, my bad...

File: view/components/Banner/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (4yki) [reply view/components/Banner/index.tsx]: Having header height depending on "window" size is bad idea. On android "window" size is different depending on StatusBar translucent prop.
https://reactnative.dev/docs/dimensions
"For Android the window dimension will exclude the size used by the status bar (if not translucent) and bottom navigation bar".
Every time we will have changes on the Statusbar it will affect header height. And calculation that are depend on this height may be broken. If you afraid of iPad UI we can just set diffrent heights for phones and tablets, but I'd not recommend using Dimensions.get("window").height for header height calcualations. The same intentions could be applied to BAnner height.

[the remark this answers is not in the document]
Comment (4yki) [reply view/components/Banner/index.tsx]: Another point, I always try to follow design guidelines. And by default react-navigation gives you header according to this guidelines. Is the any requirement for the header height, why do you want to make it windowHeight /7 ?

File: view/components/Logo.tsx (hunk 6 above)
Comment (GuyAvraham) [reply view/components/Logo.tsx:10]: ok, got u

File: view/components/header/HeaderProfileButton.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    justifyContent: 'center'
+  },
+  image: {
+    width: ComponentSize.defaultIconSize,
+    height: ComponentSize.defaultIconSize,
+    marginLeft: Margins.marginSmall
+  },
+  name: {
+    textAlign: 'right',
+    alignSelf: 'flex-end',
```
Comment (GuyAvraham) [inline view/components/header/HeaderProfileButton.tsx:28]: image

looks like if I change textAlign+ alignSelf to be center it will look better, but not perfect.
Can you set name's center to be always in the center of the image? 
And please test it with a 3-9 name length

File: view/components/Banner/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (GuyAvraham) [reply view/components/Banner/index.tsx:10]: ok cool I understand, let's stay with the fixed size...
No need to invest on tablets.

File: view/components/header/HeaderProfileButton.tsx (hunk 7 above)
Comment (GuyAvraham) [reply view/components/header/HeaderProfileButton.tsx]: I pushed a fix.

**Comment by GuyAvraham** [comment]
on the third and fourth steps: all first six words are not hidden... Here in the photo u can see only the end of it.
image
